ICSharpCode.CodeConverter/CSharp/DeclarationNodeVisitor.cs

```diff
@@ -99,8 +99,8 @@ public override async Task<CSharpSyntaxNode> VisitCompilationUnit(VBSyntax.Compi
                 .SelectAsync(async c => (UsingDirectiveSyntax) await c.Accept(TriviaConvertingVisitor));
             var usingDirectiveSyntax = usings
                 .Concat(_extraUsingDirectives.Select(u => SyntaxFactory.UsingDirective(SyntaxFactory.ParseName(u))))
-                .GroupBy(u => u.ToString())
-                .Select(g => g.First());
+                .GroupBy(u => u.Name.ToString())
+                .Select(g => g.OrderBy(v => !string.IsNullOrEmpty(v.Alias?.ToString()) ? 0 : 1).First());
```

Comment:
I think we should be grouping on name *and* alias. It's possible to alias the same import as many different things

Reply:
I'll tweak and merge this now

Reply:
I think it's a little more nuanced than that, now you mention it. I'm not sure what the right fix is without a bit more thought.

My thought behind this implementation was "if we specified an alias for a name, prefer the alias". I hadn't considered multiple aliases for the same name.

The previous implementation grouped by name, so after simplification would never use the alias (there would be two `using`s, and the aliased one would be simplified away).

Reply:
When testing this I came across a case that doesn't work. I've pushed the failing test and the tweak mentioned. I'll leave it as is to give you a chance to have a look at it, feel free to nuke my changes as always if you have a better plan

Reply:
Ah, I hadn't seen your comment when you posted. I've contrived the test to avoid the simplifier getting involved.

Reply:
Ended up down a bit of a rabbit hole here.

 * The test case you added doesn't compile in VB
 * I hit a few unrelated bugs along the way while making it compile
 * The name is unambigous when unqualified in C#, since types and names are in different namespaces

I've pushed up some changes which seem to handle this properly. I'll put together some test cases and file bugs for the things I hit along the way.

Reply:
Filed #418 for the issues I found along the way. Think this PR should be good now, unless you can think of some more unhandled cases.